src/gtfs/read.rs

```diff
@@ -3297,4 +3307,103 @@ mod tests {
             );
         });
     }
+
+    #[test]
+    fn read_gtfs_routes_as_line() {
+        fn get_collection(path: &path::Path, read_as_line: bool) -> Collections {
+            let agency_content = "agency_id,agency_name,agency_url,agency_timezone\n\
+            id_agency,My agency,http://my-agency_url.com,Europe/London";
+
+            let routes_content = "route_id,agency_id,route_short_name,route_long_name,route_type,route_color,route_text_color\n\
+                        route_1,agency_1,1,My line 1A,3,8F7A32,FFFFFF\n\
+                        route_2,agency_1,1,My line 1B,3,8F7A32,FFFFFF\n\
+                        route_4,agency_2,1,My line 1B,3,8F7A32,FFFFFF\n\
+                        route_3,agency_2,1,My line 1B,3,8F7A32,FFFFFF\n\
+                        route_5,,1,My line 1C,3,8F7A32,FFFFFF";
+
+            let trips_content =
+                "trip_id,route_id,direction_id,service_id,wheelchair_accessible,bikes_allowed\n\
+            1,route_1,0,service_1,,\n\
+            2,route_2,0,service_1,,\n\
+            3,route_3,0,service_2,,\n\
+            4,route_4,0,service_2,,\n\
+            5,route_5,0,service_2,,\n\
+            6,route_5,1,service_2,,";
+            let mut handler = PathFileHandler::new(path.to_path_buf());
+            create_file_with_content(path, "agency.txt", agency_content);
+            create_file_with_content(path, "routes.txt", routes_content);
+            create_file_with_content(path, "trips.txt", trips_content);
+            let mut collections = Collections::default();
+            let (networks, _) = super::read_agency(&mut handler).unwrap();
+            collections.networks = networks;
+            let (contributor, dataset, _) = read_utils::read_config(None::<&str>).unwrap();
+            collections.contributors = CollectionWithId::new(vec![contributor]).unwrap();
+            collections.datasets = CollectionWithId::new(vec![dataset]).unwrap();
+            super::read_routes(&mut handler, &mut collections, read_as_line).unwrap();
+            collections
+        }
+
+        test_in_tmp_dir(|path| {
+            // read route as route
+            let collections = get_collection(path, false);
+            // read route as line
+            let collections_as_line = get_collection(path, true);
```

Comment:
It is to highlight the difference between read gtfs as line instead of read gtfs as route

Reply:
You can group them in a `mod` and have tests name reflect this difference. At least, I find it very weird to do twice every line. Maybe this can be a parameterized function? Something like [this _crate_](https://crates.io/crates/parameterized) is doing (not necessarily using this _crate_ or creating macros but something like this)...